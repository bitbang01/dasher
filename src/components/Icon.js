import React from 'react'
import { Svg } from 'glamorous'
import { oneOfType, number, string } from 'prop-types'

function Icon({ size, ...props }) {
  return (
    <Svg
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      strokeLinecap="round"
      strokeLinejoin="round"
      verticalAlign="middle"
      {...props}
    />
  )
}

Icon.propTypes = {
  size: oneOfType([number, string]),
}

Icon.defaultProps = {
  size: 24,
}

export function IssueIcon(props) {
  return (
    <Icon {...props}>
      <circle cx="12" cy="12" r="10" />
      <line x1="12" y1="8" x2="12" y2="12" />
      <line x1="12" y1="16" x2="12" y2="16" />
    </Icon>
  )
}

export function PullRequestIcon(props) {
  return (
    <Icon {...props}>
      <circle cx="18" cy="18" r="3" />
      <circle cx="6" cy="6" r="3" />
      <path d="M13 6h3a2 2 0 0 1 2 2v7" />
      <line x1="6" y1="9" x2="6" y2="21" />
    </Icon>
  )
}

export function MergeIcon(props) {
  return (
    <Icon {...props}>
      <circle cx="18" cy="18" r="3" />
      <circle cx="6" cy="6" r="3" />
      <path d="M6 21V9a9 9 0 0 0 9 9" />
    </Icon>
  )
}

export function CheckIcon(props) {
  return (
    <Icon {...props}>
      <polyline points="20 6 9 17 4 12" />
    </Icon>
  )
}

export function XIcon(props) {
  return (
    <Icon {...props}>
      <line x1="18" y1="6" x2="6" y2="18" />{' '}
      <line x1="6" y1="6" x2="18" y2="18" />
    </Icon>
  )
}

export function CircleIcon(props) {
  return (
    <Icon {...props}>
      <circle cx="12" cy="12" r="10" />
    </Icon>
  )
}

export function EllipsesIcon(props) {
  return (
    <Icon {...props}>
      <circle cx="12" cy="12" r="1" />
      <circle cx="19" cy="12" r="1" />
      <circle cx="5" cy="12" r="1" />
    </Icon>
  )
}

export default Icon
