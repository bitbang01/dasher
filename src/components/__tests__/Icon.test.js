import React from 'react'
import renderer from 'react-test-renderer'
import Icon, {
  IssueIcon,
  PullRequestIcon,
  MergeIcon,
  CheckIcon,
  XIcon,
  CircleIcon,
  EllipsesIcon,
} from '../Icon'

it('renders without crashing', () => {
  const tree = renderer.create(<Icon />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders with custom color and size', () => {
  const tree = renderer.create(<Icon color="#C27E00" size={16} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe('Custom icons', () => {
  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <div>
          <IssueIcon />
          <PullRequestIcon />
          <MergeIcon />
          <CheckIcon />
          <XIcon />
          <CircleIcon />
          <EllipsesIcon />
        </div>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
