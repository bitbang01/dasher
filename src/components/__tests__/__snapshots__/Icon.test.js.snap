// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom icons renders without crashing 1`] = `
<div>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12"
      y1="16"
      y2="16"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="18"
      cy="18"
      r="3"
    />
    <circle
      cx="6"
      cy="6"
      r="3"
    />
    <path
      d="M13 6h3a2 2 0 0 1 2 2v7"
    />
    <line
      x1="6"
      x2="6"
      y1="9"
      y2="21"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="18"
      cy="18"
      r="3"
    />
    <circle
      cx="6"
      cy="6"
      r="3"
    />
    <path
      d="M6 21V9a9 9 0 0 0 9 9"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="20 6 9 17 4 12"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="18"
      x2="6"
      y1="6"
      y2="18"
    />
     
    <line
      x1="6"
      x2="18"
      y1="6"
      y2="18"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
  </svg>
  <svg
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="1"
    />
    <circle
      cx="19"
      cy="12"
      r="1"
    />
    <circle
      cx="5"
      cy="12"
      r="1"
    />
  </svg>
</div>
`;

exports[`renders with custom color and size 1`] = `
<svg
  color="#C27E00"
  fill="none"
  height={16}
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth={2}
  viewBox="0 0 24 24"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`renders without crashing 1`] = `
<svg
  fill="none"
  height={24}
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth={2}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;
